src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -290,6 +313,28 @@ private static String getCourseName(@NotNull JSONObject jsonObject, @NotNull Str
     return resourceUrls;
   }
 
+  @NotNull
+  private static List<String> getCourseAutoInstallComponentNames(@NotNull JSONObject jsonObject,
```

Comment:
please, consider making tests for this method with complex branching logics :mortar_board: